### DISCUSSION TITLE
Aiuto con Docker - come eliminare i container di HA?

### DISCUSSION THREAD

**Discussion by imxs69** [indomus/forum#3104] [HUB: Home Assistant]
Buongiorno a tutti,
utilizzo HA da qualche anno, ma sempre con installazione HAOS.
Recentemente, avendo acquistato un mini pc usato, ho installato Proxmox e creato una VM con Debian 12 dove mi sono cimentato con Docker.
Volevo provare ad installare HA in Docker e ci sono anche riuscito; il problema stava poi nella gestione di HA in Docker, così ho optato per creare una VM con HAOS e sono ripartito come prima.
Dopo un po' di mesi, ho riprovato ad utilizzare docker e mi sono creato alcuni container come Traccar, Nextcloud, Plex e Portainer ma mi sono accorto che non riesco più a cancellare i container che fanno riferimento ad HA creati mesi prima.
Killo i containers, li cancello ma dpo pochi istanti me li ritrovo nella lista dei container attivi; se li stoppo, stessa cosa, dopo pochi istanti, si riavviano.
L'unico sistema che ho trovato, è stato quelli di metterli in pausa.

Qualcuno più abile di me con Docker, potrebbe cortesemente darmi qualche dritta su come eliminarli definitivamente?

Grazie
![portainer](https://github.com/indomus/forum/assets/69854065/abf21fbc-31d5-4ffc-a51e-bfae9754af8d)

**Comment by snakuzzo** [comment]
Mi credi se ti dico che non ho capito che stai tentando di fare ?
Hai detto che sei tornato ad HA OS, giusto? Ora perché vorresti fermare e rimuovere i container che servono al funzionamento di HA OS ?
Mi sfugge qualcosa ?

**Reply by imxs69** [reply]
scusa non sono stato molto chiaro.
HA gira in una VM tutta sua direttamente sotto Proxmox mentre quelli che sto tentando di rimuovere, sono i container che avevo installato con Docker in un'altra VM Debian; questi container non mi servono e non riesco a disinstallarli, anzi quando riavvio la macchina Debian ripartono da soli creando poi problemi ad HA di produzione. 
Di tanto in tanto devo ricordarmi di controllare e verificare che siano in pausa.

Grazie

**Reply by snakuzzo** [reply]
Se magari avessi dato una minima occhiata alla documentazione di docker...

Comunque...
I container si riavviano da soli perché quando li hai creato gli hai detto tu (giustamente) di riavviarsi.
Poi...
Per eliminare un container devi...
- fermarlo: `docker stop <id o nome container>`
- rimuoverlo: `docker rm <id o nome container>`

Poi, se non ti occorre più, puoi anche rimuovere l'immagine: `docker image rm <nome immagine>`

Ora, visto che siamo leggermente off-topic, direi di chiudere qui questa discussione. Se ti occorre altro su docker, tra Google e la documentazione ufficiale puoi trovare tutto.

Ciao.

**Reply by imxs69** [reply]
Grazie per la cortese risposta, avevo comunque già cercato nella documentazione e fatto proprio quello che mi hai indicato tu, purtroppo anche se cancellati si ri-scarica l'immagine e crea nuovamente i container.

Grazie comunque

**Reply by snakuzzo** [reply]
Perché magari hai installato HA Supervised su quella VM e lo script automatico con cui fai l'installazione prevede proprio questo :D

Se ti serve una VM con Debian, fai prima a ricrearla...tempo totale -> 5 minuti

Comunque...dovesse servirti...

https://github.com/home-assistant/supervised-installer/issues/42

**Reply by imxs69** [reply]
Grazie di nuovo!

**Reply by imxs69** [reply]
Un ultimo post per ringraziarti per le dritte e per essere magari d'aiuto ad altri con il medesimo problema; seguendo un collegamento alla pagina che mi hai indicato sono giunto a
https://github.com/home-assistant/supervised-installer/issues/84

e sono riuscito a disinstallare ciò che non mi serviva, senza dover piallare Docker o senza dover rifare da capo la VM (PS avevo degli snap cmq).

Grazie di nuovo
